### PR TITLE
Add labeldrop and labelkeep actions.

### PR DESCRIPTION
**Pull request by tcolgate**
Introduce two new relabel actions. labeldrop, and labelkeep.
These can be used to filter the set of labels by matching regex

- labeldrop: drops all labels that match the regex
- labelkeep: drops all labels that do not match the regex

### CODE REVIEW

**Comment by brancz** [comment]
implementation lgtm 👍 

On a higher level we should have a general consensus as to whether this should be a feature, but personally I think it's worth it, especially for being able to protect ourselves from applications that would otherwise break Prometheus.

**Comment by tcolgate** [comment]
See the reference ticket for discussion. This doesn't help with high
cardinality metrics. It's more to do with bulk label removal to help remove
unwanted labels where they cannot easily be removed at source.

File: relabel/relabel.go [hunk 1]
```diff
@@ -88,6 +88,22 @@ func relabel(labels model.LabelSet, cfg *config.RelabelConfig) model.LabelSet {
 			}
 		}
 		labels = out
+	case config.RelabelLabelDrop:
+		out := make(model.LabelSet, 0)
```
Comment (brian-brazil) [inline relabel/relabel.go]: Make the cap here `len(labels)`, it'll be slightly faster as it'll avoid having to expand the slice.

Comment (tcolgate) [inline relabel/relabel.go]: Ok. I avoided that as I was a concerned about the potential waste of memory. Pushed update

Comment (brian-brazil) [inline relabel/relabel.go]: It's only a temporary allocation

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**